src/main/java/com/revealprecision/revealserver/service/dashboard/IrsPerformanceDashboardService.java

```diff
@@ -0,0 +1,325 @@
+package com.revealprecision.revealserver.service.dashboard;
+
+import static com.revealprecision.revealserver.constants.FormConstants.IRS_LITE_FOUND;
+import static com.revealprecision.revealserver.constants.FormConstants.IRS_LITE_NOT_SPRAYED;
+import static com.revealprecision.revealserver.constants.FormConstants.IRS_LITE_SACHET_COUNT;
+import static com.revealprecision.revealserver.constants.FormConstants.IRS_LITE_SPRAYED;
+
+import com.revealprecision.revealserver.api.v1.dto.models.ColumnData;
+import com.revealprecision.revealserver.api.v1.dto.models.RowData;
+import com.revealprecision.revealserver.constants.KafkaConstants;
+import com.revealprecision.revealserver.enums.PlanInterventionTypeEnum;
+import com.revealprecision.revealserver.messaging.message.UserLevel;
+import com.revealprecision.revealserver.messaging.message.UserPerformanceAggregate;
+import com.revealprecision.revealserver.messaging.message.UserPerformancePerDate;
+import com.revealprecision.revealserver.persistence.domain.Plan;
+import com.revealprecision.revealserver.props.DashboardProperties;
+import com.revealprecision.revealserver.props.KafkaProperties;
+import java.time.LocalDate;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.streams.StoreQueryParameters;
+import org.apache.kafka.streams.state.QueryableStoreTypes;
+import org.apache.kafka.streams.state.ReadOnlyKeyValueStore;
+import org.springframework.kafka.config.StreamsBuilderFactoryBean;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class IrsPerformanceDashboardService {
+
+  private final DashboardProperties dashboardProperties;
+  private final StreamsBuilderFactoryBean getKafkaStreams;
+  private final KafkaProperties kafkaProperties;
+  public static final String DAYS_WORKED = "Days Worked";
```

Comment:
are these constants local and only to be used by this service or may benefit from being extracted of this service?

Reply:
thanks, good spot.

I going to move the "general" performance constants to the master performance service (which calls the plan specific service) and leave the remainder in the irs specific contants in the service.
Considering that they are column names i opted to keep them in the dashboard service....